internal/controller/streams.go

```diff
@@ -196,21 +198,48 @@ func (sm *streamManager) StreamResources(ctx context.Context, logger *zap.Logger
 		logger.Error("Failed to send cluster metadata", zap.Error(err))
 		return err
 	}
+	var allWatchInfos []watcherInfo
```

Comment:
Pre-allocate the slice.
```suggestion

	allWatchInfos := make([]watcherInfo, 0, len(resourceAPIGroupMap))
```